src/frontend/staff-portal/src/app/components/jj-dispute-info/jj-dispute-updates/jj-dispute-updates.component.ts

```diff
@@ -0,0 +1,128 @@
+import { Component, Input } from '@angular/core';
+import { MatLegacyTableDataSource as MatTableDataSource } from '@angular/material/legacy-table';
+import { DisputantUpdateRequestExpanded } from '@shared/models/disputant-update-request-expanded';
+import { DisputeUpdateRequestUpdateType, Province } from 'app/api';
+import { DisputantUpdateRequest, DisputeService } from 'app/services/dispute.service';
+import { LookupsService } from 'app/services/lookups.service';
+
+@Component({
+  selector: 'app-jj-dispute-updates',
+  templateUrl: './jj-dispute-updates.component.html',
+  styleUrls: ['./jj-dispute-updates.component.scss']
+})
+export class JjDisputeUpdatesComponent {
+  @Input() disputeId: number;
+
+  dataSource = new MatTableDataSource<DisputantUpdateRequestExpanded>();
+  displayedColumns: string[] = [
+    "requestedTs",
+    "fieldTitle",
+    "oldValue",
+    "newValue",
+    "status",
+    "statusTs"
+  ];
+  DISPUTANT_ADDRESS_FIELDS = {
+    "AddressLine1":"Address Line 1", 
+    "AddressLine2":"Address Line 2", 
+    "AddressLine3":"Address Line 3", 
+    "AddressCity":"City", 
+    "AddressProvince":"Province", 
+    "AddressCountry":"Country",
+    "PostalCode":"Postal Code"
+  };
+  DISPUTANT_PHONE_FIELDS = {
+    "HomePhoneNumber":"Phone Number"
+  };
+  DISPUTANT_NAME_FIELDS = {
+    "DisputantGivenName1":"Disputant Given Name 1", 
+    "DisputantGivenName2":"Disputant Given Name 2", 
+    "DisputantGivenName3":"Disputant Given Name 3", 
+    "DisputantSurname":"Surname"
+  };
+
+  constructor(private disputeService: DisputeService, private lookupsService: LookupsService) {
+  }
+
+  ngOnInit(): void {
+    this.disputeService.getDisputeUpdateRequests(this.disputeId).subscribe(request => {
+      let data : DisputantUpdateRequestExpanded[] = [];
+      // expand the DisputeUpdateRequest object so that each entry in updateJson is a separate row in the MAT table
+      request.forEach(request => {
+        // The updateJson structure contains ALL Disputant fields, not just the ones pertaining to the specific updateType
+        // So, we first need to extract the fields pertaining to the specific updateType and then use only those fields
+        // in the MAT table.
+        let updateJson = JSON.parse(request.updateJson);
```

Comment:
Is there a type for updateJson ? If so, it can be typed.

Reply:
No, I don't think there is, at least I'm not aware of a type. In ORDS, it's just a string field that stores an arbitary json structure ([here](https://github.com/bcgov/jag-traffic-courts-online/blob/28dd2557d2c522fa84c93cea5d70dd998cb6bf49/src/backend/oracle-data-api/src/main/resources/occm-openapi-spec.yaml#L1321)). 

As a result, when this field shows up as string type in the API.